solidity/contracts/hooks/layer-zero/LayerZeroV2Hook.sol

```diff
@@ -0,0 +1,135 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+/*@@@@@@@       @@@@@@@@@
+ @@@@@@@@@       @@@@@@@@@
+  @@@@@@@@@       @@@@@@@@@
+   @@@@@@@@@       @@@@@@@@@
+    @@@@@@@@@@@@@@@@@@@@@@@@@
+     @@@@@  HYPERLANE  @@@@@@@
+    @@@@@@@@@@@@@@@@@@@@@@@@@
+   @@@@@@@@@       @@@@@@@@@
+  @@@@@@@@@       @@@@@@@@@
+ @@@@@@@@@       @@@@@@@@@
+@@@@@@@@@       @@@@@@@@*/
+import {MessagingParams, MessagingFee, ILayerZeroEndpointV2} from "@layerzerolabs/lz-evm-protocol-v2/contracts/interfaces/ILayerZeroEndpointV2.sol";
+import {Message} from "../../libs/Message.sol";
+import {TypeCasts} from "../../libs/TypeCasts.sol";
+import {Indexed} from "../../libs/Indexed.sol";
+import {IPostDispatchHook} from "../../interfaces/hooks/IPostDispatchHook.sol";
+import {AbstractMessageIdAuthHook} from "../libs/AbstractMessageIdAuthHook.sol";
+import {StandardHookMetadata} from "../libs/StandardHookMetadata.sol";
+
+struct LayerZeroV2Metadata {
+    /// @dev the endpoint Id. prev dstChainId
+    uint32 eid;
+    /// @dev if the source transaction is cheaper than the amount of value passed, refund the additional amount to this address
+    address refundAddress;
+    /// @dev  parameters for the adapter service, e.g. send some dust native token to dstChain. prev adapterParam
+    bytes options;
+}
+
+contract LayerZeroV2Hook is AbstractMessageIdAuthHook, Indexed {
```

Comment:
`Indexed` is only needed if we are emitting events that the Relayer or Validator needs to index
in this case I dont think its necessary

Reply:
fixed